lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -567,3 +610,32 @@ def _fail_action_with_error_log(self, event, action: str, message: str) -> None:
     def _defer_action_with_info_log(self, event, action: str, message: str) -> None:
         logger.info("Deferring %s: %s", action, message)
         event.defer()
+
+    def _success_action_with_info_log(self, event, action: str, results: Dict[str, str]) -> None:
+        logger.info("%s completed successfully", action.capitalize())
+        event.set_results(results)
+
+    def _log_backup_restore_result(self, current_pbm_status, previous_pbm_status) -> None:
+        operation_result = self._get_backup_restore_operation_result(
+            current_pbm_status, previous_pbm_status
+        )
+        logger.info(operation_result)
+
+    def _get_backup_restore_operation_result(self, current_pbm_status, previous_pbm_status) -> str:
+        if (
+            type(current_pbm_status) == type(previous_pbm_status)
+            and current_pbm_status.message == previous_pbm_status.message
+        ):
+            return f"Operation is still in progress: '{current_pbm_status.message}'"
+
+        if (
```

Comment:
some comments would be useful here - specifically to explain why you check the prev status instead of current

(it makes sense to me, but took me awhile to figure it out 😅 )

Reply:
Done